tests/unittests/algo/test_space.py

```diff
@@ -451,55 +364,28 @@ def test_that_objects_types_are_ok(self):
         assert ['2', 'asdfa'] not in dim
         assert [2, 'asdfa'] in dim
 
-    def test_no_default_value(self):
-       categories = {'asdfa': 0.1, 2: 0.2, 3: 0.3, 'lalala': 0.4}
-       dim = Categorical('yolo', categories)
-       assert dim.default_value == None
-
-    def test_cast_to_dimension_type(self):
+    def test_init_with_default_value_string(self):
         categories = {'asdfa': 0.1, 2: 0.2, 3: 0.3, 'lalala': 0.4}
-        dim = Categorical('yolo', categories)
+        dim = Categorical('yolo', categories, default_value="asdfa")
 
-        v = dim.cast_to_dimension_type(1)
-        sample = dim.sample()[0]
-        
-        assert type(v) == type(sample)
+        assert type(dim.default_value) is str
 
-    @pytest.mark.xfail(reason="type implementation detail")
-    def test_init_with_default_value(self):
+    def test_init_with_default_value_int(self):
         categories = {'asdfa': 0.1, 2: 0.2, 3: 0.3, 'lalala': 0.4}
-        dim = Categorical('yolo', categories, default_value = "def")
-        sample = dim.sample()[0]
+        dim = Categorical('yolo', categories, default_value=2)
 
-        assert type(dim.default_value) == type(sample)
+        assert type(dim.default_value) is int
 
-    def test_set_none_default_value(self):
+    def test_init_with_wrong_default_value(self):
         categories = {'asdfa': 0.1, 2: 0.2, 3: 0.3, 'lalala': 0.4}
-        dim = Categorical('yolo', categories)
-        dim.default_value = None
-
-        assert dim.default_value == None
-        assert str(dim) == "Categorical(name=yolo, " \
-                           "prior={asdfa: 0.10, 2: 0.20, 3: 0.30, lalala: 0.40}, " \
-                           "shape=(), default value=None)"
+        
+        with pytest.raises(ValueError):
+            dim = Categorical('yolo', categories, default_value=2.3)
```

Comment:
This is now also tested in test_set_outside_bounds_default_value no? Actually I think test_set_outside_bounds_default_value does not make much sense for Categorical